merkle/tree_hasher.go

```diff
@@ -14,11 +14,30 @@
 
 package merkle
 
-// TreeHasher is the interface that the previous tree hasher struct implemented.
-type TreeHasher interface {
-	HashEmpty() []byte
+// LogHasher provides the hash functions needed to compute dense merkele trees.
+type LogHasher interface {
```

Comment:
The interfaces currently have a lot in common. Are they going to diverge in future PRs. If not is there value in having them share a common base to highlight the log / map differences?